src/stateMachine.tsx

```diff
@@ -68,8 +69,16 @@ export function createStore(
 
   setUpDevTools(isDevMode, storageType, getName, getStore);
 
-  if (result && Object.keys(result).length) return;
-  setStore(data);
+  if (result && Object.keys(result).length) {
+    Object.keys(result).forEach(key => {
```

Comment:
any chance just do a shallow merge, so we wouldn't need `lodash` merge?
```
{
  ...result,
  ...data,
}
```

Reply:
alternative:

`setStore(result && Object.keys(result).length ? result : data);`
or
`setStore(result && Object.keys(result).length ? {...result, ...data} : data);`

Reply:
``` 
if (result && Object.keys(result).length) {
    Object.keys(result).forEach(key => {
      if (Object.keys(data).includes(key)) {
         data[key] = { ...data[key], ...result[key] };
      }
    });
  } 
 setStore(data);

```

Reply:
This way { form: {a: 'a'} } and { form: {b: 'b'} } will merge nicely